### ISSUE TITLE
Evaluate False incorrect assumption

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#21145]
```
from sympy.core.parameters import evaluate
a=Integer(-3)
with evaluate(False):
    print(a.is_extended_negative) # gives False
```

**Comment by oscargus** [comment]
This seems fixed in current master.

**Comment by oscarbenjamin** [comment]
It was fixed in 91f3ad35d2610c6dbbcede95bb15e2dba074f5be from #20721. A test for pretty much exactly this case was added then.
